### DISCUSSION TITLE
Interactions wont work after bot restarts

### DISCUSSION THREAD

**Discussion by PureGolded** [Pycord-Development/pycord#1525] [Q&A] [answered]
Hi,

I am trying to make a ticket tool bot and to create a ticket I have a message with buttons. However, when I make a modification to the bot and restart it, I can no longer use the buttons and I have to resend the message, which is not ideal since I am planning on making the bot public.

So my question is if I can somehow restart the bot without having to resend every button interaction.

code:
`class TicketView(discord.ui.View):
    def __init__(self):
        super().__init__(timeout=None)

        supportServerButton = discord.ui.Button(label="How to..", style=discord.ButtonStyle.url, url='**********')
        self.add_item(supportServerButton)
    
    @discord.ui.button(label="Create", style=discord.ButtonStyle.green)
    async def button_callback(self, button, inter):
        await inter.response.send_message("Ticket created!", delete_after=5)
        try:
            category = discord.utils.get(inter.guild.categories, id=*********)
        except:
            print("No category found")
            inter.response.send_message("Couldn't create a ticket, an error occurred", delete_after=10)
            return
        channel = await category.create_text_channel(f"ticket_{inter.user.name}")
        message = await channel.send(f"User {inter.user.display_name} created a ticket!\nuser id: {inter.user.id}")
        userembed = discord.Embed(title="Ticket created", description=f"see it [here]({message.jump_url})")
        await inter.user.send(embed=userembed)
        pin_message = await channel.send(view=TicketEditClose())
        await pin_message.pin()
        
        global permissions
        permissions = category.overwrites_for(inter.guild.default_role)
        
        permissions.send_messages=True
        permissions.view_channel=True
        await channel.set_permissions(inter.guild.default_role, overwrite=permissions)`

I tried to search for my answer everywhere, but I couldn't find it. 
Thanks in advance for any help.

**Comment by JustaSqu1d** [comment, marked as answer]
You can try to use Persistent views which are explained [here](https://guide.pycord.dev/interactions/ui-components/buttons#persistent-views) and examples are [here](https://github.com/Pycord-Development/pycord/blob/master/examples/views/persistent.py).

**Reply by PureGolded** [reply]
Yes, Thank you so much I didn't scroll down enough.
